### PR TITLE
ci: cleanup e2e tests and builds from jenkins combined job

### PR DESCRIPTION
**Pull request by yakimant**
No e2e are going to be run on ci.infra. Cleaning up the combined job.

### CODE REVIEW

File: ci/tests/Jenkinsfile.e2e-nightly [hunk 1]
```diff
@@ -23,15 +33,14 @@ pipeline {
...
   stages {
     stage('Fetch') {
-      when { expression { !params.APK_URL } }
       steps { script {
         copyArtifacts(
-          projectName: "status-mobile/nightly",
-          filter: '*-x86.apk',
-          /* WARNING: This copies the latest available artifact. */
-          selector: lastWithArtifacts(),
+          projectName: "status-mobile/platforms/android-e2e",
```
Comment (jakubgs) [inline ci/tests/Jenkinsfile.e2e-nightly:38]: I'm not sure this path makes sense, since the `android-e2e` build is no longer part of the `combined` job.

**Review by jakubgs — APPROVED** [review]
Looks good, just not sure about the path for e2e build.

File: ci/tests/Jenkinsfile.e2e-nightly (hunk 1 above)
Comment (yakimant) [reply ci/tests/Jenkinsfile.e2e-nightly:38]: Ah, that's because we use another "combined" job to run nightly e2e - `ci/Jenkinsfile.e2e-nightly`

Comment (yakimant) [reply ci/tests/Jenkinsfile.e2e-nightly:38]: Let me double check with CI test run:
https://ci.status.im/job/status-mobile/job/e2e-nightly/70/

Comment (yakimant) [reply ci/tests/Jenkinsfile.e2e-nightly:38]: All good

**Pull request merged**